modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -118,6 +132,14 @@ ICache<K, V> getDiskCache() {
         return diskCache;
     }
 
+    // Package private for testing.
+    void enableDisableDiskCache(Boolean isDiskCacheEnabled) {
+        // When disk cache is disabled, we are not clearing up the disk cache entries yet as that should be part of
+        // separate cache/clear API.
```

Comment:
>With this logic where the cleanup is not happening on disabling, we will end up in a situation where a key exists in both onHeap and onDisk cache.

Yes this can happen. We expect the users to explicitly clear the cache after doing this. 
I am assuming you were using cleanup here similar to clearing up whole cache(and not cleanup via invalidation).

Regarding cleanup logic to remove stale keys, this will still end up clearing stale keys in disk cache as it eventually calls `invalidate(key)` where we don't check whether disk cache is enabled or not.

Reply:
We should document these behavior then and it seems more of an expert setting in that case. Do you have any documentation issue tracking these settings ?

Reply:
Yes, we need to document this. 
Yes, will add it as part of https://github.com/opensearch-project/documentation-website/issues/6300

Reply:
Have updated the description `Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)` to reflect that